docs/plugins/versioning/ui/components/NavbarDropdown.js

```diff
@@ -0,0 +1,88 @@
+import React, { useState, useEffect } from 'react';
+import styles from './NavbarDropdown.module.css';
+
+const CACHE_KEY = 'kargo-docs-versions';
+const CACHE_DURATION = 1000 * 60 * 30; // 30 minutes
+const githubApiUrl = 'https://api.github.com/repos/akuity/kargo/branches';
+
+function NavbarDropdown() {
+  console.log("Navbar dropdown initialized");
+  const [versions, setVersions] = useState([]);
+  const [loading, setLoading] = useState(true);
+  const [currentVersion, setCurrentVersion] = useState("");
+
+  useEffect(() => {
+    fetchVersions();
+  }, []);
+
+  const fetchVersions = async () => {
+    try {
+      console.log("Before fetching versions")
+      const response = await fetch(githubApiUrl);
```

Comment:
Shouldn't you look in local storage instead of unconditionally making this request?